src/Microsoft.AspNetCore.Mvc.Core/Internal/ControllerActionDescriptorBuilder.cs

```diff
@@ -119,16 +119,14 @@ public static IList<ControllerActionDescriptor> Build(ApplicationModel applicati
                     // selected when a route group returned by the route.
                     if (hasAttributeRoutes)
                     {
-                        actionDescriptor.RouteConstraints.Add(new RouteDataActionConstraint(
-                            TreeRouter.RouteGroupKey,
-                            string.Empty));
+                        actionDescriptor.RouteValues.Add(TreeRouter.RouteGroupKey, string.Empty);
                     }
 
                     // Add a route constraint with DenyKey for each constraint in the set to all the
                     // actions that don't have that constraint. For example, if a controller defines
                     // an area constraint, all actions that don't belong to an area must have a route
                     // constraint that prevents them from matching an incoming request.
-                    AddRemovalConstraints(actionDescriptor, removalConstraints);
+                    AddRemovalConstraints(actionDescriptor, routeValueKeys);
```

Comment:
`AddRouteValue` - also the comment above needs to be updated. `DenyKey` no longer exists.